ncm-symlink/src/main/perl/symlink.pod

```diff
@@ -28,92 +25,148 @@ Target definition and contextual variable value may also contain expression inte
-             "target",   "/bin/tcsh",
-             "exists",    true
-            )
-      );
-
-    "/software/components/symlink/links" = 
-      list(nlist(
-             "name",    "/atlas",
-             "target",   "/atlas_prod/@@uname@@",
-             "exist",    true
-            )
-      );
-
-    "/software/components/symlink/context" = 
-      list(nlist(
-             "name",    "ostype",
-             "value",   "@@uname@@",
-            )
-      );
-    "/software/components/symlink/links" = 
-      list(nlist(
-             "name",    "/atlas",
-             "target",   "/atlas_prod/{ostype}",
-             "exists",    true
-            )
-      );
+C<exists>: (boolean) check that the target exists when creating it or check that the symlink name exists when deleting it.
 
-    # Allow replacement of empty directories and links
-    "/software/components/symlink/options/replace/dirempty" = ".saved"; 
-    "/software/components/symlink/options/replace/link" = "yes";
+=item *
+
+C<replace>: (nlist) option used to specify the action to take when an object with the same name as the symlink already exists, depending on the object type. 
+Possible actions are: do not define the symlink, replace the
+object by the symlink or define the symlink after renaming the object. The nlist keys and values can be:
+
+=over
+
+=item *
+
+key: the existing object type. Valid values are: C<all>, C<dir>, C<dirempty>, C<file>, C<link>, C<none>.  C<dirempty> means an empty directory only, C<dir> means any directory.
+C<all> and C<none> are mutually exclusive but can be combined with other object types to define the extension to use when renaming a given object type or to prevent/enable replacement for a specific object type. 
+
+=item *
 
-    "/software/components/symlink/links" = 
-      list(nlist(
-             "name",    "/usr/local",
-             "target",   "/lal_prod/{ostype}",
-             "exists",    true
-            )
-      );
+value: action applying to the object type. Can be C<yes> (replacement of the object by the symlink allowed), C<no> (replacement of the object by the symlink disabled) or any other
+string. In this latter case, replacement of the object by the symlink is enabled after renaming the object by appending the string to its name. The value can also be empty: see
+below. Note that non empty directories are B<always> renamed before defining the symlink (a default extension, C<.ncm-symlink_saved>, is used).
 
-=head1 DEPENDENCIES
+=back
 
-None.
+=back
 
-=head1 BUGS
+C<replace> option allows a lot of flexibility in specifying what should be done in case of conflict with an existing object. It implements the following advanced features:
 
-None known.
+=over
 
-=head1 AUTHOR
+=item *
 
-Michel Jouvin <jouvin@lal.in2p3.fr>
+C<none=extension> can be used to establish a default rename extension without enabling replacement. This extension will be used with object types for which replacement is enabled.
```

Comment:
the `without enabling replacement`  vs `for which replacement is enabled` is still a bit unclear. you mean it's a default and thus you don't need to specify it each time you enable replacement?